### PR TITLE
Task consumer integration

### LINKED ISSUE #819: [BUG] Document task consumer integration
Issue:  https://github.com/opensearch-project/OpenSearch/issues/1009
PR:  https://github.com/opensearch-project/OpenSearch/pull/2293

### PR DESCRIPTION
**Pull request by JeffH-AWS**
### Description
Added additional objects to the JSON responses in the task API page and added a description of the feature under the logs page.

### Issues Resolved
Fixes #819

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by JeffH-AWS** [comment]
Pinged the developer of this feature for a technical review.  I want to add a little bit more information about the log settings for task resource consumer in the opensearch.yml file.  But that isn't a blocker to getting the rest tech reviewed.  Will follow up once the dev has had a chance to look.  It's late in the day so it might not be reviewed until tomorrow AM (8/11)

File: _opensearch/rest-api/tasks.md [hunk 1]
```diff
@@ -49,7 +49,13 @@ Note that if a task finishes running, it won't be returned as part of your reque
           "start_time_in_millis": 1613599752458,
           "running_time_in_nanos": 994000,
           "cancellable": false,
-          "headers": {}
+          "headers": {},
+          "resource_stats" : {
+            "total" : {
+              "cpu_time_in_nanos" : 0,
+              "memory_in_bytes" : 0
+            }
```
Comment (sruti1312) [inline _opensearch/rest-api/tasks.md:57]: These fields will only be present if detailed flag is set and will not be included as a part of default response. This behaviour is similar to ```description``` field which is not present in this sample response and hence I think we should not include these fields as well.

**Comment by JeffH-AWS** [comment]
Got clarification on the `resource_stats` object - it will only be returned in the JSON if the `detailed` flag is passed.  Additionally, the `detailed` flag will also expose a `description` field.  Sruti is going to provide a sample command and sample response which I'll append to the API doc.

**Comment by sruti1312** [comment]
Sample command and response with detailed flag. 

```
curl -XGET "localhost:9200/_tasks?actions=*search&detailed&pretty
```

```
{
  "nodes" : {
    "CRqNwnEeRXOjeTSYYktw-A" : {
      "name" : "runTask-0",
      "transport_address" : "127.0.0.1:9300",
      "host" : "127.0.0.1",
      "ip" : "127.0.0.1:9300",
      "roles" : [
        "cluster_manager",
        "data",
        "ingest",
        "remote_cluster_client"
      ],
      "attributes" : {
        "testattr" : "test",
        "shard_indexing_pressure_enabled" : "true"
      },
      "tasks" : {
        "CRqNwnEeRXOjeTSYYktw-A:677" : {
          "node" : "CRqNwnEeRXOjeTSYYktw-A",
          "id" : 677,
          "type" : "transport",
          "action" : "indices:data/read/search",
          "description" : "indices[], search_type[QUERY_THEN_FETCH], source[{\"query\":{\"query_string\":{\"query\":\"king\",\"fields\":[],\"type\":\"best_fields\",\"default_operator\":\"or\",\"max_determinized_states\":10000,\"enable_position_increments\":true,\"fuzziness\":\"AUTO\",\"fuzzy_prefix_length\":0,\"fuzzy_max_expansions\":50,\"phrase_slop\":0,\"analyze_wildcard\":false,\"escape\":false,\"auto_generate_synonyms_phrase_query\":true,\"fuzzy_transpositions\":true,\"boost\":1.0}}}]",
          "start_time_in_millis" : 1660106254525,
          "running_time_in_nanos" : 1354236,
          "cancellable" : true,
          "cancelled" : false,
          "headers" : { },
          "resource_stats" : {
            "total" : {
              "cpu_time_in_nanos" : 0,
              "memory_in_bytes" : 0
            }
          }
        }
      }
    }
  }
}

```

File: _opensearch/logs.md [hunk 2]
```diff
@@ -168,7 +168,36 @@ node1 | [2019-10-24T19:48:51,012][WARN][i.i.s.index] [node1] [some-index/i86iF5k
 
 Slow logs can consume considerable disk space if you set thresholds or levels too low. Consider enabling them temporarily for troubleshooting or performance tuning. To disable slow logs, return all thresholds to `-1`.
 
+## Task logs
+
+OpenSearch can log CPU time and memory utilization for the top N expensive search tasks when task resource consumers are enabled. By default, task resource consumers will log the top 10 search tasks at 60 second intervals. These values can be configured in `opensearch.yml`.
+
+Task logging can be enabled dynamically through the cluster settings API:
```
Comment (lizsnyder) [inline _opensearch/logs.md:175]: Nit: should probably be active voice

**Review by Naarcha-AWS — APPROVED** [review]

File: _opensearch/logs.md (hunk 2 above)
Comment (JeffH-AWS) [reply _opensearch/logs.md]: Reworded it a bit

File: _opensearch/logs.md [hunk 3]
```diff
@@ -168,7 +168,36 @@ node1 | [2019-10-24T19:48:51,012][WARN][i.i.s.index] [node1] [some-index/i86iF5k
 
 Slow logs can consume considerable disk space if you set thresholds or levels too low. Consider enabling them temporarily for troubleshooting or performance tuning. To disable slow logs, return all thresholds to `-1`.
 
+## Task logs
+
+OpenSearch can log CPU time and memory utilization for the top N expensive search tasks when task resource consumers are enabled. By default, task resource consumers will log the top 10 search tasks at 60 second intervals. These values can be configured in `opensearch.yml`.
```
Comment (sruti1312) [inline _opensearch/logs.md:173]: Can we rephrase to ```top N memory expensive search tasks``` to make it more accurate?

File: _opensearch/rest-api/tasks.md [hunk 4]
```diff
@@ -70,7 +71,7 @@ Note that if a task finishes running, it won't be returned as part of your reque
           "start_time_in_millis": 1613599750929,
           "running_time_in_nanos": 1529733100,
           "cancellable": true,
-          "headers": {}
+          "headers": {},
```
Comment (sruti1312) [inline _opensearch/rest-api/tasks.md:74]: We can revert this change?

File: _opensearch/rest-api/tasks.md [hunk 5]
```diff
@@ -142,6 +143,60 @@ GET /_tasks?nodes=opensearch-node1
 }
 ```
 
+The following request will return details about active search tasks:
```
Comment (sruti1312) [inline _opensearch/rest-api/tasks.md:146]: Maybe also include that this will return detailed information?

File: _opensearch/rest-api/tasks.md [hunk 6]
```diff
@@ -142,6 +143,60 @@ GET /_tasks?nodes=opensearch-node1
...
+        "testattr" : "test",
+        "shard_indexing_pressure_enabled" : "true"
+      },
+      "tasks" : {
+        "CRqNwnEeRXOjeTSYYktw-A:677" : {
+          "node" : "CRqNwnEeRXOjeTSYYktw-A",
+          "id" : 677,
+          "type" : "transport",
+          "action" : "indices:data/read/search",
+          "description" : "indices[], search_type[QUERY_THEN_FETCH], source[{\"query\":{\"query_string\":{\"query\":\"king\",\"fields\":[],\"type\":\"best_fields\",\"default_operator\":\"or\",\"max_determinized_states\":10000,\"enable_position_increments\":true,\"fuzziness\":\"AUTO\",\"fuzzy_prefix_length\":0,\"fuzzy_max_expansions\":50,\"phrase_slop\":0,\"analyze_wildcard\":false,\"escape\":false,\"auto_generate_synonyms_phrase_query\":true,\"fuzzy_transpositions\":true,\"boost\":1.0}}}]",
```
Comment (sruti1312) [inline _opensearch/rest-api/tasks.md:180]: Can we not include the actual query string and instead replace with $QUERY_STRING?

File: _opensearch/logs.md (hunk 3 above)
Comment (JeffH-AWS) [reply _opensearch/logs.md:173]: Will do!

File: _opensearch/rest-api/tasks.md (hunk 6 above)
Comment (JeffH-AWS) [reply _opensearch/rest-api/tasks.md]: Done

File: _opensearch/rest-api/tasks.md (hunk 5 above)
Comment (JeffH-AWS) [reply _opensearch/rest-api/tasks.md:146]: Done

File: _opensearch/logs.md [hunk 7]
```diff
@@ -168,7 +168,36 @@ node1 | [2019-10-24T19:48:51,012][WARN][i.i.s.index] [node1] [some-index/i86iF5k
 
 Slow logs can consume considerable disk space if you set thresholds or levels too low. Consider enabling them temporarily for troubleshooting or performance tuning. To disable slow logs, return all thresholds to `-1`.
 
+## Task logs
+
+OpenSearch can log CPU time and memory utilization for the top N memory expensive search tasks when task resource consumers are enabled. By default, task resource consumers will log the top 10 search tasks at 60 second intervals. These values can be configured in `opensearch.yml`.
```
Comment (kolchfa-aws) [inline _opensearch/logs.md:173]: memory-expensive

**Review by kolchfa-aws — APPROVED** [review]
LGTM

**Comment by JeffH-AWS** [comment]
Leaving open/unmerged until 2.2 is cut.
